### PR TITLE
[MN-1057] Validate provisioning profile list before save

### PR DESCRIPTION
**Pull request by slapec93**

### CODE REVIEW

File: models/common.go [hunk 1]
```diff
@@ -12,3 +12,24 @@ type Record struct {
 	CreatedAt time.Time `json:"created_at" gorm:"created_at" db:"created_at"`
 	UpdatedAt time.Time `json:"updated_at" gorm:"updated_at"`
 }
+
```
Comment (zormandi) [inline models/common.go:15]: I really dislike the idea of having to write this sort of functionality ourselves. I know there's no built-in support for it in the language but why don't we use some lib like https://github.com/thoas/go-funk ?

File: models/app_settings.go [hunk 2]
```diff
@@ -23,6 +23,26 @@ func (s IosSettings) Valid() bool {
 	return !reflect.DeepEqual(s, IosSettings{})
 }
 
+// ValidateProvisioningProfileSlugs ...
+func (s *IosSettings) ValidateSelectedProvisioningProfileSlugs(provProfiles []string) {
+	if len(provProfiles) == 0 && len(s.SelectedAppStoreProvisioningProfiles) > 0 {
+		s.SelectedAppStoreProvisioningProfiles = []string{}
+		return
+	}
+	for _, slug := range s.SelectedAppStoreProvisioningProfiles {
```
Comment (zormandi) [inline models/app_settings.go:32]: This is basically an intersection operation between two slices. Again, let's use a lib for this.

File: services/app_settings_patch.go [hunk 3]
```diff
@@ -79,9 +80,20 @@ func AppSettingsPatchHandler(env *env.AppEnv, w http.ResponseWriter, r *http.Req
...
+func prepareAppSettingsToUpdate(api bitrise.APIInterface, appSettingsToUpdate *models.AppSettings, params AppSettingsPatchParams) (*models.AppSettings, []string, error) {
 	updateWhiteList := []string{}
 	if params.IosSettings.Valid() {
+		if len(params.IosSettings.SelectedAppStoreProvisioningProfiles) > 0 {
+			existingProvProfiles, err := api.GetProvisioningProfiles(appSettingsToUpdate.App.BitriseAPIToken, appSettingsToUpdate.App.AppSlug)
+			if err != nil {
+				return nil, []string{}, err
+			}
+			existingProvProfileSlugs := []string{}
+			for _, provProfile := range existingProvProfiles {
```
Comment (zormandi) [inline services/app_settings_patch.go:92]: This is a `map` operation. :)

**Review by zormandi — CHANGES_REQUESTED** [review]

File: models/app_settings.go (hunk 2 above)
Comment (komaromia) [reply models/app_settings.go:33]: So this is not in the funk lib?

**Review by komaromia — APPROVED** [review]

**Pull request merged**